### PR TITLE
Do simple logging for failed export.

### PR DESCRIPTION
**Pull request by carlosalberto**
We already log other situations (such as failing to do a clean shutdown), so it makes sense to add logging for failed export.

### CODE REVIEW

**Comment by iNikem** [comment]
Will that solve #1480 ?

**Comment by carlosalberto** [comment]
@iNikem You referenced this same PR ;)

**Comment by iNikem** [comment]
Bah! I mean #1418

**Comment by carlosalberto** [comment]
I think this should fix #1418 yes, mostly because we are already logging the other important failure case (failure to properly shutdown).

The rest of the errors do not seem unrecoverable (they are all parameters-handling related), so I think this simple PR is enough.

File: exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java [hunk 1]
```diff
@@ -126,6 +126,7 @@ public ResultCode export(Collection<SpanData> spans) {
       stub.postSpans(request);
       return ResultCode.SUCCESS;
     } catch (Throwable e) {
+      logger.log(Level.SEVERE, "Failed to export spans", e);
```
Comment (anuraaga) [inline exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java:129]: We don't seem to use `SEVERE` anywhere else yet - spans can fail to export for transient reasons (backend is restarting), and observability is not critical to the app anyways, so I'd go with WARNING if not INFO.

Comment (iNikem) [reply exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java:129]: Please at least WARN :)

**Comment by iNikem** [comment]
@anuraaga Taking into account https://github.com/open-telemetry/opentelemetry-specification/issues/716, I would say logging in each exporter makes more sense :)

**Comment by anuraaga** [comment]
@iNikem Well I'm definitely ok with removing resultcode instead :) But it seems like one or the other makes sense, either exporter is left to their own ways or are controlled by the span processor.

File: exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java (hunk 1 above)
Comment (carlosalberto) [reply exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java:129]: WARN works for me. Will update.

**Comment by carlosalberto** [comment]
On top of the possibility of having the export result code being removed, there's the case of using a `MultiSpanProcessor` and having a specific backend fail (imagine that Jaeger exported its data fine, but Zipkin didn't). Also, in each processor's export step you have access to the actual `Exception`, which *could be* useful.

**Comment by anuraaga** [comment]
`MultiSpanProcessor` is also a `SpanProcessor`, if we think that's the right layer to log exporter failures it could take care of it too (sorry wasn't clear with my reference, I linked to batch but simple, multi would also probably have a similar change if going with it).

> Also, in each processor's export step you have access to the actual Exception, which could be useful.

The exception could be propagated up too - it seems weird to return a failure status without the reason. Shouldn't we either leave everything to the exporter, or otherwise have it propagate everything? I personally think either is fine, but a mix seems strange, it's hard to know who has responsibility when it comes to exporting spans.

**Comment by carlosalberto** [comment]
>  Shouldn't we either leave everything to the exporter

Going forward, we should 'normalize' things with Exception/Exporter Code being returned or being totally gone. For the time being, I suggest we keep these warnings and tune this part once we decide how to proceed (most likely after we come to an agreement on whether we want #716 or not).

**Pull request merged**